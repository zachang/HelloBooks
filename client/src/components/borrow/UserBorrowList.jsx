import React from 'react';
import PropTypes from 'react-proptypes';
import { Link } from 'react-router';

const userBorrowList = props => (
  <tr>
    <td>{ props.borrower.Book.bookName }</td>
    <td>{ props.borrower.Book.Category.categoryName }</td>
    <td>{ props.borrower.User.fullname }</td>
    <td>{ new Date(props.borrower.createdAt).toLocaleString() }</td>
    <td>{ props.borrower.returned}</td>
    <td>
      { ((props.borrower.borrowStatus === 'true') || (props.clickedBorrowList.indexOf(props.borrower.id) > -1)) ? 'Confirmed' :
        <Link
          className='waves-effect waves-light btn btn-small'
          onClick={() => (props.confirmBorrow(props.borrower.id))}
        >
          Confirm
        </Link>
      }
    </td>
  </tr>
);

userBorrowList.propTypes = {
  borrower: PropTypes.object.isRequired,

};
export default userBorrowList;
