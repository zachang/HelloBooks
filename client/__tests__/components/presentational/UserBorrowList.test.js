import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';

import UserBorrowList from '../../../src/components/borrow/UserBorrowList.jsx';

describe('<UserBorrowList/>', () => {
  const props = {
    borrower: {
      Book: {
        Category: {
          categoryName: 'Science'
        }
      },
      User: {
        fullname: 'Tito Navo'
      }
    },
    confirmBorrow: () => '',
    clickedBorrowList: []
  };

  const shallowWrapper = shallow(<UserBorrowList {...props}/>);
  const tree = toJson(shallowWrapper);
  let wrapper;

  const wrapperMount = mount(<UserBorrowList {...props} />);

  it('renders <UserBorrowList /> component', () => {
    expect(tree).toMatchSnapshot();
  });

  it('calls onClick', () => {
    wrapper = shallow(<UserBorrowList {...props} />);
    wrapper.find('.btn-small').simulate('click');
    expect(wrapperMount.find('.btn-small').at(0).text()).toBe('Confirm');
  });
});

