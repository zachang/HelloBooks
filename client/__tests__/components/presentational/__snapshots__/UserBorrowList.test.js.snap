// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserBorrowList/> renders <UserBorrowList /> component 1`] = `
<tr>
  <td />
  <td>
    Science
  </td>
  <td>
    Tito Navo
  </td>
  <td>
    Invalid Date
  </td>
  <td />
  <td>
    <Link
      className="waves-effect waves-light btn btn-small"
      onClick={[Function]}
      onlyActiveOnIndex={false}
      style={Object {}}
    >
      Confirm
    </Link>
  </td>
</tr>
`;
